### PR TITLE
#1, adds isPortReady function that creates and destroys sockets until…

### PR DESCRIPTION
**Pull request by JustinBeaudry**
… a connection is made, adds options.waitTTL (defaults to 10000ms)

### CODE REVIEW

File: wat.js [hunk 1]
```diff
@@ -151,4 +151,42 @@ function many (character, group) {
   return Array(count + 1).join(character);
 }
 
+var isPortReady = function(port, TTL, next) {
```
Comment (bevacqua) [inline wat.js]: Just do `function isPortReady (port, ttl, next) {`

Comment (bevacqua) [inline wat.js]: Also, move it into its own module file.

File: wat.js [hunk 2]
```diff
@@ -151,4 +151,42 @@ function many (character, group) {
...
+
+  function checkPort() {
+    if (status === 'open') {
+      if (timer) {
+        clearInterval(timer);
+      }
+      if (timeout) {
+        clearTimeout(timeout);
+      }
+      return next();
```
Comment (bevacqua) [inline wat.js]: `next(); return;`

File: wat.js [hunk 3]
```diff
@@ -151,4 +151,42 @@ function many (character, group) {
...
+    }
+    var socket = new net.Socket();
+    socket.on('connect', function() {
```
Comment (bevacqua) [inline wat.js]: name these methods, `function connect () {`, etc.

**Comment by JustinBeaudry** [comment]
@bevacqua Updated.

**Pull request merged**